tools/mo/openvino/tools/mo/moc_frontend/pipeline.py

```diff
@@ -215,7 +218,6 @@ def shape_to_array(shape: PartialShape):
 
     ngraph_function = moc_front_end.convert(input_model)
 
-    # TO DO: remove as part of PyTorch frontend productization CVS-103615
     if argv.framework == "pytorch":
-        pytorch_process_after_convert(argv, ngraph_function)
+        resolve_input_signature(argv, ngraph_function)
```

Comment:
not sure why signature (input order, etc.) resolving can not be handled under the hood of PT FE? Looks not correct.

Reply:
this info does not provided inside model and from discussion with @mvafin , I know that it is impossible to add tensor names to InputModel

Reply:
Looks like we can (and we should) move Pytorch specific code, like this one, to PT FE with extending Decoder API. Decoder is what connects original model with the conversion core, that means that we can provide any auxiliary information for model conversion pre- and post-processing via decoder. The problem which is specific for PT FE is that this pre- and post-processing should happens in Python. This is not an issue for us, because we can extend Decoder with pre- and post-processing methods with appropriate arguments and call them from fe.convert method implemented in C++. Actually it is also related to tracing/scripting model -- it also can be a part of the FE, not general MO code. @eaidova , do you feel confident at changing API in this way or you prefer someone else to sketch it for you?

Reply:
@slyalin, scripting moved to decoder, input signature adding if applicable too, please review

Reply:
can we hide resolving of the input signature in `moc_front_end.convert()` method? Can we provide specific implementation of Python part of `convert` method for PT FE? I think input signature resolving should be a part of conversion.

Reply:
no, we need to know it before conversion, because user can provide named input information via inputs which will be failed if input names from signature will be not available from input_model